src/python/aqueduct_executor/operators/function_executor/execute.py

```diff
@@ -237,18 +228,19 @@ def run(spec: FunctionSpec) -> None:
                 # Cast np.bool_ to a bool.
                 result = bool(result)
             else:
-                handle_type_error_and_exit(spec, storage, exec_state, TIP_NOT_BOOL)
+                raise ExecFailureException(
```

Comment:
I don't know if this is crucial, but it seems that we will lose all user printed logs for such failures?

Reply:
We won't lose any user logs. The except clause will add the user logs to the exec_state before writing. `exec_state.from_exception()` forces you to add user logs.